lib/vrl/stdlib/src/del.rs

```diff
@@ -166,14 +167,23 @@ impl Expression for DelFn {
     }
 
     fn type_def(&self, _: &state::Compiler) -> TypeDef {
-        TypeDef::new().unknown()
+        TypeDef::any()
     }
 
     fn update_state(
         &mut self,
         state: &mut state::Compiler,
     ) -> std::result::Result<(), ExpressionError> {
-        self.query.delete_type_def(state);
+        // FIXME(Jean): This should also delete non-external queries, as `del(foo.bar)` is
+        // supported.
```

Comment:
Note that the `if self.query.is_external()` isn't new, I only pulled it up a level, out of the `delete_type_def` method. This note is merely a reminder that we need to fix this. I've since created an issue at #11246.